homeassistant/components/stookwijzer/sensor.py

```diff
@@ -23,44 +65,40 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up Stookwijzer sensor from a config entry."""
-    client = hass.data[DOMAIN][entry.entry_id]
-    async_add_entities([StookwijzerSensor(client, entry)], update_before_add=True)
+
+    async_add_entities(
+        StookwijzerSensor(description, entry) for description in STOOKWIJZER_SENSORS
+    )
 
 
-class StookwijzerSensor(SensorEntity):
-    """Defines a Stookwijzer binary sensor."""
+class StookwijzerSensor(CoordinatorEntity, SensorEntity):
+    """Defines a Stookwijzer sensor."""
 
-    _attr_attribution = "Data provided by stookwijzer.nu"
-    _attr_device_class = SensorDeviceClass.ENUM
+    entity_description: StookwijzerSensorDescription
+    _attr_attribution = "Data provided by atlasleefomgeving.nl"
     _attr_has_entity_name = True
-    _attr_name = None
-    _attr_translation_key = "stookwijzer"
-
-    def __init__(self, client: Stookwijzer, entry: ConfigEntry) -> None:
-        """Initialize a Stookwijzer device."""
-        self._client = client
-        self._attr_options = [cls.value for cls in StookwijzerState]
-        self._attr_unique_id = entry.entry_id
+
+    def __init__(
+        self,
+        description: StookwijzerSensorDescription,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize the Sensor Entity."""
+        self.entity_description = description
+        self._coordinator = entry.runtime_data
+
+        super().__init__(self._coordinator)
+
+        self._attr_unique_id = f"{entry.entry_id}{DOMAIN}{description.key}"
         self._attr_device_info = DeviceInfo(
-            identifiers={(DOMAIN, f"{entry.entry_id}")},
-            name="Stookwijzer",
-            manufacturer="stookwijzer.nu",
+            identifiers={(DOMAIN, entry.entry_id)},
+            manufacturer="Atlas Leefomgeving",
             entry_type=DeviceEntryType.SERVICE,
-            configuration_url="https://www.stookwijzer.nu",
+            configuration_url="https://www.atlasleefomgeving.nl/stookwijzer",
         )
 
-    def update(self) -> None:
-        """Update the data from the Stookwijzer handler."""
-        self._client.update()
-
-    @property
-    def available(self) -> bool:
-        """Return if entity is available."""
-        return self._client.state is not None
-
     @property
     def native_value(self) -> str | None:
         """Return the state of the device."""
-        if self._client.state is None:
-            return None
-        return StookwijzerState(self._client.state).value
+        value = self.entity_description.value_fn(self._coordinator)
+        return str(value) if value is not None else value
```

Comment:
We should not stringify everything

Reply:
Yup, this code no longer exists in my local version, new version will just return value_fn